updated mako from 1.0.7 to 1.2.2

zenmonkeykstop opened the pull request:
## Status

Ready for review.

## Description of Changes

Updates `mako`, an `alembic` dependency, from v1.0.7 to 1.2.2


## Testing

- [x] verify that CI is passing
- start a dev container on this branch with `cd securedrop/; ./bin/dev-shell`
- in the container run `cd securedrop; source ./bin/dev-deps && maybe_create_config_py`
- run the following commands to create an empty database:
  ```
  sudo mkdir -p /var/lib/securedrop/
  sudo chmod 0700 /var/lib/securedrop/
  sudo chown "$(id -u):$(id -g)" /var/lib/securedrop
  rm -f /var/lib/securedrop/db.sqlite
  sqlite3 /var/lib/securedrop/db.sqlite .databases > /dev/null
  ```
- [x] verify that the command `alembic upgrade head` completes successfully
- [x] verify that the command `alembic history` completes successfully
- [x] verify that you can successfully downgrade to earlier alembic revisions and then upgrade again
- [x] try creating a new migration (`./bin/new-migration 'my migration message'`), verify it looks reasonable

## Deployment
 
n/a

## Checklist

### If you made non-trivial code changes:

- [x] I have written a test plan and validated it for this PR

Choose one of the following:

- [ ] I have opened a PR in the [docs repo](https://github.com/freedomofpress/securedrop-docs) for these changes, or will do so later
- [ ] I would appreciate help with the documentation
- [x] These changes do not require documentation

### If you added or updated a production code dependency:

Choose one of the following:

- [x] I have performed a diff review and pasted the contents: https://github.com/freedomofpress/securedrop-debian-packaging/wiki/mako-1.0.7-to-1.2.2

```
$ gpg --verify mako.asc
gpg: Signature made Tue Sep  6 10:03:04 2022 EDT
gpg:                using RSA key A05ACE4F7E0BCA9BAD8D42F4ACFE4353173015EA
gpg: Good signature from "Kevin O'Gorman <kog@stunbunny.org>" [unknown]
gpg:                 aka "Kevin O'Gorman <kevin.ogorman@gmail.com>" [unknown]
gpg:                 aka "Kevin O'Gorman <kog@freedom.press>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: A05A CE4F 7E0B CA9B AD8D  42F4 ACFE 4353 1730 15EA
```

zenmonkeykstop committed:
updated mako from 1.0.7 to 1.2.2

legoktm commented on securedrop/requirements/python3/securedrop-app-code-requirements.in:
```diff
@@ -11,6 +11,7 @@ Flask-SQLAlchemy
 Flask-WTF>=1.0.0
 Flask>=2.0.2
 Jinja2>=3.0.0
+mako>=1.2.2
```
I would prefer we didn't add indirect dependencies to the requirements.in file, as it makes it confusing in the future whether a dependency is actually needed by us when upstreams change their dependencies.

Now that requirements.txt has been bumped, you should be able to remove it from requirements.in, re-run the update thing, and it'll fix the `via` to be just alembic, but keep it at 1.2.2.

legoktm replied:
@zenmonkeykstop: I removed the mako line from this file, re-ran `make update-python3-dependencies` twice, and both times it did not downgrade `mako`, so I think it's good now.

legoktm committed:
Remove mako from securedrop-app-code-requirements.in

This is just a transitive dependency of alembic, so we don't want
to make it look like we actually depend on it directly..

legoktm reviewed (approved):
Going to wait until @zenmonkeykstop can look at my changes before I merge it. I also verified generating a new migration file works properly, which is the main usage of mako AIUI.

zenmonkeykstop reviewed (commented):
Looks good to me @legoktm - merge at will!
